src/validation/__tests__/UniqueArgumentDefinitionNamesRule-test.ts

```diff
@@ -0,0 +1,120 @@
+import { describe, it } from 'mocha';
+
+import { UniqueArgumentDefinitionNamesRule } from '../rules/UniqueArgumentDefinitionNamesRule';
+
+import { expectSDLValidationErrors } from './harness';
+
+function expectSDLErrors(sdlStr: string) {
+  return expectSDLValidationErrors(
+    undefined,
+    UniqueArgumentDefinitionNamesRule,
+    sdlStr,
+  );
+}
+
+function expectValidSDL(sdlStr: string) {
+  expectSDLErrors(sdlStr).to.deep.equal([]);
+}
+
+describe('Validate: Unique argument definition names', () => {
+  it('no args', () => {
+    expectValidSDL(`
+      type SomeObject {
+        someField: String
+      }
+
+      interface SomeInterface {
+        someField: String
+      }
+
+      directive @someDirective on QUERY
+    `);
+  });
+
+  it('one argument', () => {
+    expectValidSDL(`
+      type SomeObject {
+        someField(foo: String): String
+      }
+
+      interface SomeInterface {
+        someField(foo: String): String
+      }
+
+      directive @someDirective(foo: String) on QUERY
+    `);
+  });
+
+  it('multiple arguments', () => {
+    expectValidSDL(`
+      type SomeObject {
+        someField(
+          foo: String
+          bar: String
+        ): String
+      }
+
+      interface SomeInterface {
+        someField(
+          foo: String
+          bar: String
+        ): String
+      }
+
+      directive @someDirective(
+        foo: String
+        bar: String
+      ) on QUERY
+    `);
+  });
+
+  it('duplicating arguments', () => {
+    expectSDLErrors(`
+      type SomeObject {
+        someField(
+          foo: String
+          bar: String
+          foo: String
```

Comment:
Do we require a test where the same argument is defined twice with different types?